framework/web/Session.php

```diff
@@ -935,10 +938,22 @@ protected function freeze()
      */
     protected function unfreeze()
     {
-        $this->open();
         if (null !== $this->frozenSessionData) {
-            $_SESSION = $this->frozenSessionData;
-            $this->frozenSessionData = null;
+
+            YII_DEBUG ? session_start() : @session_start();
+
+            if ($this->getIsActive()) {
+                Yii::info('Session unfrozen', __METHOD__);
+            } else {
+                $error = error_get_last();
+                $message = isset($error['message']) ? $error['message'] : 'Failed to unfreeze session.';
+                Yii::error($message, __METHOD__);
+            }
+
+            if ($this->frozenSessionData !== null) {
```

Comment:
It seems that this condition is redundant. It is already wrapped with the same `if` condition and the code above does not change `frozenSessionData` variable, is it?